tz.cpp

```diff
@@ -218,7 +218,12 @@ static std::string get_install()
 
 #ifndef INSTALL
 
-static const std::string install = get_install();
+static std::string install = get_install();
+
+void set_install(std::string install_folder)
+{
+   install = install_folder;
+}
```

Comment:
I think it is up to the client caller to ensure that the install path is set before date time library functionality is invoked (including tzdata initialization). I can't imagine anyone wanting to change the path of tzdata during program execution, after tzdata has been loaded. As a library caller I'd be very happy with some thread safety documentation added here for a function meant to set the install path: https://howardhinnant.github.io/date/tz.html#Reference. Even without that documentation I would expect client callers to infer that install path has to be set before tzdata is loaded.

The restrictions we face are:

- Our executables don't know where their install paths are until runtime. Our users can install our executables wherever they wish.

- We disabled auto download functionality, which could affect performance in unpredictable ways. We can't guarantee that users have access to the built-in download folder. We also don't want recurring downloads because our product can run inside Docker images.

- Our code base has 80+ entry points that will set the install path of the running executables. We don't want to spend time figuring out which entry points are hit when, so whenever an entry point sets the product's install path, there we pass the install path into the date time libraries, so the path setting function in question here needs to be re-entrant.